webapp/app.py

```diff
@@ -60,8 +60,75 @@ def global_template_context():
         "child_paths": child_paths
     }
 
+from ruamel.yaml import YAML
+import json
+
+_yaml = YAML(typ="rt")
+_yaml_safe = YAML(typ="safe")
+
+def get_yaml_loader(typ="safe"):
+    if typ == "safe":
+        return _yaml_safe
+    return _yaml
+
+def get_file(filename, replaces={}):
+    """
+    Reads a file, replaces occurences of all the keys in `replaces` with
+    the correspondant values and returns the resulting string or None
+
+    Keyword arguments:
+    filename -- name if the file to load.
+    replaces -- key/values to replace in the file content (default {})
+    """
+    filepath = os.path.join(flask.current_app.root_path, filename)
+
+    try:
+        with open(filepath, "r") as f:
+            data = f.read()
+            for key in replaces:
+                data = data.replace(key, replaces[key])
+    except Exception:
+        data = None
+
+    return data
+
+def get_yaml(filename, typ="safe", replaces={}):
+    """
+    Reads a file, replaces occurences of all the keys in `replaces` with the
+    correspondant values and returns an ordered dict with the YAML content
+
+    Keyword arguments:
+    filename -- name if the file to load.
+    typ -- type of yaml loader
+    replaces -- key/values to replace in the file content (default {})
+    """
+    try:
+        yaml = get_yaml_loader(typ)
+        data = get_file(filename, replaces)
+        return yaml.load(data)
+    except Exception:
+        return None
+
+def examples_list_view():
+    examples = get_yaml("../docs/examples/examples.yaml")
+
+    context = {
+        "examples": examples,
+    }
+
+    return flask.render_template("examples/index.html", **context)
+
+
+def example_view(subpath):
+    examples = get_yaml("../docs/examples/examples.yaml")
+    path = subpath.split("/")
+
+    example = examples[path[0]][path[1]]
```

Comment:
This obviously needs a better implementation to support more nested examples and to 404 on paths that are not in YAML...